frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -0,0 +1,775 @@
+                </Grid>
+            </Grid>
+
+            <Grid row className={"modal-info-row"}>
+                <Grid className={"modal-info-label"}>Result:</Grid>
+                <Grid className={`modal-info-value ${successClass}`}>
+                    {dataItem.connectionCheckSuccessful ? "success" : "failed"}
+                </Grid>
+            </Grid>
+
+            <Grid row className={"modal-info-row"}>
+                <Grid className={"modal-info-label"}>Started At:</Grid>
+                <Grid className={"modal-info-value"}>
+                    {formatDate(dataItem.connectionCheckStartedAt)}
+                    <br />
+                    {dataItem.connectionCheckStartedAt}
+                </Grid>
+            </Grid>
+
+            <Grid row className={"modal-info-row"}>
+                <Grid className={"modal-info-label"}>Time to complete:</Grid>
+                <Grid className={"modal-info-value"}>
+                    {duration()}
+                    <br />
+                </Grid>
+            </Grid>
+
+            <Grid row className={"modal-info-row"}>
+                <Grid className={"modal-info-label"}>Result message:</Grid>
+                <Grid className={"modal-info-value"}>
+                    {dataItem.connectionCheckResult}
+                </Grid>
+            </Grid>
+        </GridContainer>
+    );
+}
+
+export function AdminReceiverDashboard() {
+    const [startDate, setStartDate] = useState<string>(defaultStartDateIso());
+    const [endDate, setEndDate] = useState<string | undefined>(); // may be null because it's optional
+    const results = useResource(AdmConnStatusResource.list(), {
+        startDate,
+        endDate,
+    });
+    // this is the text input box filter
+    const [filterReceivers, setFilterReceivers] = useState("");
+    const [filterErrorResults, setFilterErrorResults] = useState("");
+    const [filterRowSuccessState, setFilterRowSuccessState] =
+        useState<SuccessRate>(SuccessRate.UNDEFINED);
+
+    // used to show hide the modal
+    const modalShowInfoRef = useRef<ModalRef>(null);
+    const [currentDataForModal, setCurrentDataForModal] = useState<
+        AdmConnStatusDataType[]
+    >([]);
+
+    const data = makeDictionary(
+        results.filter(
+            (eachRow) =>
+                eachRow.filterOnName(filterReceivers) &&
+                eachRow.filterOnCheckResultStr(filterErrorResults)
+        )
+    );
+
+    const showDetailsModal = (subData: AdmConnStatusDataType[]) => {
+        if (subData.length) {
+            setCurrentDataForModal(subData);
+            modalShowInfoRef?.current?.toggleModal(undefined, true);
+        }
+    };
+
+    return (
+        <section className="grid-container">
+            <h4>Receiver Status dashboard</h4>
+            <section>
+                CRON job results that check if receivers are working.
+            </section>
+            <form autoComplete="off" className="grid-row margin-0">
```

Comment:
I know you mentioned most of this not being reusable, but these date and string based filters could potentially be really useful in other views. Not important to split these out now, but wanted to plant the seed of an idea